embassy-net/src/lib.rs

```diff
@@ -823,9 +825,11 @@ impl<D: Driver> Inner<D> {
 
         // Apply DNS servers
         #[cfg(feature = "dns")]
-        s.sockets
-            .get_mut::<smoltcp::socket::dns::Socket>(self.dns_socket)
-            .update_servers(&dns_servers[..]);
+        if !dns_servers.is_empty() {
+            s.sockets
+                .get_mut::<smoltcp::socket::dns::Socket>(self.dns_socket)
+                .update_servers(&dns_servers[..DNS_MAX_SERVER_COUNT]);
```

Comment:
This should probably use https://docs.rs/smoltcp/latest/smoltcp/config/constant.DNS_MAX_SERVER_COUNT.html instead.

Reply:
Yes, thank you, I missed this one.

Reply:
`dns_servers[..DNS_MAX_SERVER_COUNT]`

this will crash if max is 2 but DHCP gives us 1.